examples/llama.vim

```diff
@@ -671,15 +733,61 @@ function! s:fim_on_stdout(job_id, data, event) dict
     endif
 
     " display the suggestion and append the info to the end of the first line
-    call nvim_buf_set_extmark(l:bufnr, l:id_vt_fim, s:pos_y - 1, s:pos_x - 1, {
-        \ 'virt_text': [[s:content[0], 'llama_hl_hint'], [l:info, 'llama_hl_info']],
-        \ 'virt_text_win_col': virtcol('.') - 1
-        \ })
+    if s:nvim_ghost_text
+        call nvim_buf_set_extmark(l:bufnr, l:id_vt_fim, s:pos_y - 1, s:pos_x - 1, {
+            \ 'virt_text': [[s:content[0], 'llama_hl_hint'], [l:info, 'llama_hl_info']],
+            \ 'virt_text_win_col': virtcol('.') - 1
+            \ })
 
-    call nvim_buf_set_extmark(l:bufnr, l:id_vt_fim, s:pos_y - 1, 0, {
-        \ 'virt_lines': map(s:content[1:], {idx, val -> [[val, 'llama_hl_hint']]}),
-        \ 'virt_text_win_col': virtcol('.')
-        \ })
+        call nvim_buf_set_extmark(l:bufnr, l:id_vt_fim, s:pos_y - 1, 0, {
+            \ 'virt_lines': map(s:content[1:], {idx, val -> [[val, 'llama_hl_hint']]}),
+            \ 'virt_text_win_col': virtcol('.')
+            \ })
+    elseif s:vim_ghost_text
+        " adapted from:
+        " https://github.com/github/copilot.vim/blob/release/autoload/copilot.vim
+        let l:new_suffix = s:content[0]
+        let l:current_suffix = getline('.')[col('.') - 1 :]
+        let l:inset = ''
+        while !empty(l:new_suffix)
+            let last_char = matchstr(l:new_suffix, '.$')
+            let l:new_suffix = matchstr(l:new_suffix, '^.\{-\}\ze.$')
+            if last_char ==# matchstr(l:current_suffix, '.$')
+                if !empty(l:inset)
+                    call prop_add(line('.'), col('.') + len(l:current_suffix), {
+                                \ 'type': s:hlgroup,
+                                \ 'text': l:inset
+                                \ })
+                    let l:inset = ''
+                endif
+                let l:current_suffix = matchstr(l:current_suffix, '^.\{-\}\ze.$')
+            else
+                let l:inset = last_char . l:inset
+            endif
+        endwhile
+        if !empty(l:new_suffix . l:inset)
+            call prop_add(line('.'), col('.'), {
+                        \ 'type': s:hint_hlgroup,
+                        \ 'text': l:new_suffix . l:inset
+                        \ })
+        endif
+        for line in s:content[1:]
+            call prop_add(line('.'), 0, {
+                        \ 'type': s:hint_hlgroup,
+                        \ 'text': l:new_suffix . line,
+                        \ 'text_padding_left': s:get_indent(line),
```

Comment:
This padding does not seem to be necessary:

```suggestion
```

Reply:
![snip57](https://github.com/user-attachments/assets/49de133b-d5f4-488f-afe3-6f70858bf226)
For me, it seems to be required

Reply:
Hm, here it looks like this with the padding:

<img width="504" alt="image" src="https://github.com/user-attachments/assets/600cb9ee-fc4e-46ff-b42c-2afa8d06fe5d">

And without the padding:

<img width="503" alt="image" src="https://github.com/user-attachments/assets/3bed8223-e696-4843-9619-6d1129bb3c93">

Reply:
I think I'm forgetting to address the tabstop. What do you get as output when you execute `:echo &shiftwidth | :echo &tabstop`? What are your vimrc settings regarding tabbing/whitespace?

Reply:
```
:echo &shiftwidth | :echo &tabstop
4
4
```

This is my [.vimrc](https://github.com/ggerganov/ggterm/blob/master/.vimrc)

Reply:
I think it's fixed now